source/GameInterface/Services/MapEvents/SettlementEntered.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Common.Messaging;
+using ProtoBuf;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameInterface.Services.MapEvents
+{
+    /// <summary>
+    /// Event fired when the local player enters a settlement.
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public record SettlementEntered : IEvent
```

Comment:
For any network events, I've been prefixing with `Network`
```suggestion
    public record NetworkPartySettlementEntered : IEvent
```